### PR TITLE
Support specifying zip archive fetching of bundled libzmq source in ZMQ_PREFIX

### PR DESCRIPTION
**Pull request by fox4u**
```
# zip archives
ZMQ_PREFIX=https://github.com/zeromq/libzmq/archive/ecc63d0.zip

# git clone (default branch latest)
ZMQ_PREFIX=git@https://github.com/zeromq/libzmq

# git clone and checkout some ref
ZMQ_PREFIX=git@https://github.com/zeromq/libzmq@ecc63d0
```

### CODE REVIEW

File: buildutils/bundle.py [hunk 1]
```diff
@@ -161,6 +169,87 @@ def fetch_libzmq(savedir):
...
+    if url.endswith('.zip'):
+        fetch_libzmq_repo_zip(savedir, url)
+    else:
+        try:
+            repo = Repo(dest)
+        except (InvalidGitRepositoryError, NoSuchPathError):
+            info("invalid local repo, clone from %s" % url)
+            if os.path.exists(dest):
+                shutil.rmtree(dest, ignore_errors=False, onerror=handle_remove_readonly)
+            repo = Repo.clone_from(url, dest)
```
[the remark this answers is not in the document]
Comment (fox4u) [reply buildutils/bundle.py:213]: I was thinking git cli output might be unstable, and could be affected by different versions, platforms, or even users' locale. Without this build-time dependency we may need to do a lot more to handle compatibility and error conditions, as in GitPython itself. 

Another option is just dropping git support, if it is nearly never used. Moreover we now have an official libzmq 4.3.5 release to bundle. We may keep the archive zip fetching though, as it also does the job, and if you are OK with it. Please advice.

[the remark this answers is not in the document]
Comment (minrk) [reply buildutils/bundle.py:213]: I don't think git output is parsed, so I don't think there's any output to handle.

i'm okay with dropping git support if the archive is good enough for you.

[the remark this answers is not in the document]
Comment (fox4u) [reply buildutils/bundle.py]: Archive is definitely good enough. Let's keep it sweet and simple. My PR branch is updated as per your review comments.

**Comment by minrk** [comment]
Thanks!

**Pull request merged**